### PR TITLE
prov/verbs: fix overwrite of return value

### PR DESCRIPTION
**Pull request by rhalkyard**
Fix for issue #3357 

Signed-off-by: Richard Halkyard <rhalkyard@cray.com>

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c [hunk 1]
```diff
@@ -261,8 +261,8 @@ static int fi_ibv_rdm_av_insert(struct fid_av *av_fid, const void *addr,
 			   ntohs(av_entry->addr.sin_port), av_entry);
 
 		av->used++;
-		ret++;
 	}
+	ret = count - failed;
```
Comment (a-ilango) [inline prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c]: `failed` is incremented in error case only when `(av->flags & FI_EVENT)` is true so `count - failed` may be more than the actual # of addresses inserted.

Comment (rhalkyard) [reply prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c]: Good point; I'm not hugely familiar with the verbs provider codebase and must have missed that. I've pushed another commit that always increments `failed` on error. That way, we count insertions and failures the same way, whether `FI_EVENT` is set or not.

**Comment by shefty** [comment]
Looks good to me.  @a-ilango - please double check

**Comment by a-ilango** [comment]
This looks good to me. It seems the previous implementation also had the bug of incorrectly reporting # of successful insertions since ret was tracking that and the function was returning ret - failed, essentially double counting the # of failures.

**Pull request merged**